src/parse.coffee

```diff
@@ -1,44 +1,44 @@
 
+deckardcain = require('deckardcain')
 fury = require('fury')
 fury.use(require('fury-adapter-apib-parser'))
 fury.use(require('fury-adapter-swagger'))
 
 
-furyParse = (source, options, callback) ->
-  [callback, options] = [options, {}] if typeof options is 'function'
+createAnnotation = (message, type) ->
+  {
+    element: 'annotation'
+    meta: {classes: [type]}
+    content: message
+  }
 
-  args = {source, generateSourceMap: true}
-  args.mediaType = 'text/vnd.apiblueprint' if options.forceApiBlueprint
 
-  fury.parse(args, (err, result) ->
-    if not (err or result)
+parse = (source, callback) ->
+  annotations = []
+  mediaType = deckardcain.identify(source)
+
+  unless mediaType
+    mediaType = 'text/vnd.apiblueprint'
+    annotations.push(createAnnotation('''\
+      Could not recognize API description format. \
+      Falling back to API Blueprint by default.\
+    ''', 'warning'))
+
+  args = {source, mediaType, generateSourceMap: true}
```

Comment:
To detect transaction examples (see `test/unit/detect-transaction-examples-test.coffee`) which were present in AST but aren't used in API Elements, but Dredd uses them for addressing transactions in hooks (to be changed with https://github.com/apiaryio/dredd/issues/227).